### PR TITLE
[Draft] Introduce RemoveNoEffectSplitPass on luci/pass

### PR DESCRIPTION
**Pull request by struss**
For #5456 

This commit will introduce RemoveNoEffectSplitPass.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci/pass/src/RemoveNoEffectSplitPass.test.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+TEST(RemoveNoEffectSplitPass, remove_no_effect_split)
+{
+  auto graph = loco::make_graph();
+  create_no_effect_split_graph(graph.get());
+
+  luci::RemoveNoEffectSplitPass pass;
+  while (pass.run(graph.get()))
+    ;
+  luci::CircleSplit *split_node = nullptr;
+  int count = 0;
```
Comment (jinevening) [inline compiler/luci/pass/src/RemoveNoEffectSplitPass.test.cpp:68]: `count` seems unnecessary in this test.

File: compiler/luci/pass/src/RemoveNoEffectSplitPass.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{
+  auto target_node = dynamic_cast<luci::CircleSplitOut *>(node);
+  if (target_node == nullptr)
+    return false;
+
+  auto split_node = dynamic_cast<luci::CircleSplit *>(target_node->input());
+  if (split_node == nullptr)
+    return false;
+
+  if (loco::succs(split_node).size()!=1)
```
Comment (jinevening) [inline compiler/luci/pass/src/RemoveNoEffectSplitPass.cpp:33]:
```suggestion
  if (loco::succs(split_node).size() != 1)
```

File: compiler/luci/pass/src/RemoveNoEffectSplitPass.test.cpp (hunk 1 above)
Comment (struss) [reply compiler/luci/pass/src/RemoveNoEffectSplitPass.test.cpp]: Okay I will fix it.

File: compiler/luci/pass/src/RemoveNoEffectSplitPass.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci/pass/src/RemoveNoEffectSplitPass.cpp]: Okay I will fix it.

**Comment by struss** [comment]
Done!

**Pull request closed without merging**